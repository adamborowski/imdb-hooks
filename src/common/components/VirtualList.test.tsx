import getVirtualList, {IVirtualListProps, VirtualList} from './VirtualList';
import React, {ComponentType} from 'react';
import {ListChildComponentProps} from 'react-window';
import {create} from 'react-test-renderer';

const Row: ComponentType<ListChildComponentProps> = p => <div>row #{p.index}</div>;

const MyVirtualList = getVirtualList<any>();

const createList = (props?: Partial<IVirtualListProps<any>>, List = MyVirtualList) => (
  <List row={Row} itemSize={100} {...props} />
);

describe('VirtualList', () => {
  it('renders properly', () => {
    const list = create(createList({ padding: 1 })).toJSON();

    expect(list).toMatchSnapshot();
    expect(list).toHaveStyleRule('padding', '1px 0 1px 1px');
  });
});
