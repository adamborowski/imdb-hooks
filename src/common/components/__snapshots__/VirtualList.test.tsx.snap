// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualList renders properly 1`] = `
.c1::-webkit-scrollbar {
  width: 22px;
}

.c1::-webkit-scrollbar-thumb {
  border: 6px solid transparent;
  border-radius: 30px;
  background-clip: padding-box;
  min-height: 30px;
  background-color: rgba(0,0,0,0.15);
}

.c0 {
  width: 100%;
  height: 100%;
  background: #ffffff;
  padding: 1px 0 1px 1px;
}

.c0 .fixed-size-list-container {
  width: 100%;
  height: 100%;
}

.c0 .react-window > * > * {
  width: calc(100% - 0.5px) !important;
  min-width: 500px;
}

<div
  className="c0"
>
  <div
    className="fixed-size-list-container ant-list ant-list-split"
  >
    <div
      className="react-window c1"
      onScroll={[Function]}
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "direction": "ltr",
          "height": 0,
          "overflow": "auto",
          "position": "relative",
          "width": -0.5,
          "willChange": "transform",
        }
      }
    >
      <div
        style={
          Object {
            "height": 0,
            "pointerEvents": "",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
